flocker/ca/functional/test_validation.py

```diff
@@ -158,15 +178,7 @@ def _handshake(self, credential):
                 server_context_factory = validating_context_factory
                 client_context_factory = peer_context_factory
 
-            server_endpoint = SSL4ServerEndpoint(reactor, port,
-                                                 server_context_factory,
-                                                 interface='127.0.0.1')
-            server_factory = WaitForDisconnectsFactory.forProtocol(
-                SendingProtocol
-            )
-            self.addCleanup(lambda: server_factory.wait_for_disconnects())
-            d = server_endpoint.listen(server_factory)
-            d.addCallback(lambda port: self.addCleanup(port.stopListening))
+            start_tls_server(self, port, server_context_factory)
```

Comment:
I guess this assumes that server endpoints synchronously listen... which isn't quite what the interface guarantees... though the TLS endpoint happens to do this.  Pre-existing, anyway.  File an issue if you don't fix it here.